session/session_test.go

```diff
@@ -3141,6 +3141,31 @@ func (s *testSessionSuite2) TestStmtHints(c *C) {
 	c.Assert(tk.Se.GetSessionVars().GetReplicaRead(), Equals, kv.ReplicaReadFollower)
 }
 
+// Test MEMORY_QUOTA hint in POINT_GET/BATCH_POINT_GET
+func (s *testSessionSuite2) TestPointGetStmtHints(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1;")
+	tk.MustExec("create table t1(a int);")
+	defer config.RestoreFunc()()
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.OOMAction = config.OOMActionCancel
+	})
+	tk.MustExec("insert /*+ MEMORY_QUOTA(1 GB) */ into t1 values (1);")
+	tk.MustExec("select /*+ MEMORY_QUOTA(0 GB) */ * from t1 where a = 1;")
+	val := int64(0)
+	c.Assert(tk.Se.GetSessionVars().StmtCtx.MemTracker.CheckBytesLimit(val), IsTrue)
+	c.Assert(tk.Se.GetSessionVars().StmtCtx.GetWarnings()[0].Err.Error(), Equals, "Setting the MEMORY_QUOTA to 0 means no memory limit")
+	tk.MustExec("set tidb_mem_quota_query=1;")
+	err := tk.QueryToErr("select * from t1 where a = 1;")
```

Comment:
I have a test, and the sql will use TableReader, not PointGet.
So we should create a table using primary key, and build a plan using PointGet/BatchPointGet?

Reply:
We can use `explain select * from t1 where a = 1;` to know the detail of the plan.

Reply:
I get it,thanks!

Reply:
I get it,thanks!